packages/tools/kadena-cli/src/account/commands/accountFund.ts

```diff
@@ -78,66 +64,56 @@ export const createAccountFundCommand = createCommand(
+    if (networkConfig.networkId.includes('mainnet')) {
+      return log.error(
+        `Fundings are not allowed on "${networkConfig.networkId}" network.`,
       );
-      return;
     }
 
     if (accountConfig.fungible.trim() !== 'coin') {
-      log.error(`\nYou can't fund an account other than "coin" fungible.\n`);
-      return;
+      return log.error(`You can't fund an account other than "coin" fungible.`);
     }
 
-    // if (networkConfig.networkId === 'development') {
-    //   if (!(await networkIsAlive(networkConfig.networkHost))) {
-    //     console.log(
-    //       chalk.red(
-    //         `\nDevnet host "${networkConfig.networkHost}" is not running.\n`,
-    //       ),
-    //     );
-    //     return;
-    //   }
-
-    //   const hasModuleAvailable = await describeModule(FAUCET_MODULE_NAME, {
-    //     host: networkConfig.networkHost,
-    //     defaults: {
-    //       networkId: networkConfig.networkId,
-    //       meta: { chainId: chainId },
-    //     },
-    //   }).catch(() => false);
-
-    //   if (hasModuleAvailable === false) {
-    //     console.log(
-    //       chalk.yellow(
-    //         `\nFaucet module is not available on chain "${chainId}" in "${networkConfig.network}".\n`,
-    //       ),
-    //     );
-
-    //     const { deployDevnet } = await option.deployDevnet();
-
-    //     if (!deployDevnet) {
-    //       return;
-    //     }
-
-    //     console.log('\nDeploying faucet...\n');
-
-    //     await deployDevNetFaucet([chainId]).catch((e) => {
-    //       console.log(
-    //         chalk.red(
-    //           `\nFailed to deploy faucet module on chain "${chainId}" in "${network}".\n`,
-    //         ),
-    //       );
-    //       throw Error(e);
-    //     });
-    //     console.log(
-    //       chalk.green(
-    //         `\nDeployed faucet module on chain "${chainId}" in "${network}".\n`,
-    //       ),
-    //     );
-    //   }
-    // }
+    if (networkConfig.networkId.includes('development')) {
+      if (!(await networkIsAlive(networkConfig.networkHost))) {
+        return log.error(
+          `Devnet host "${networkConfig.networkHost}" is not running.`,
+        );
+      }
+
+      const undeployedChainIds = await findMissingModuleDeployments(
+        FAUCET_MODULE_NAME,
+        networkConfig,
+        chainId,
+      );
+
+      const undeployedChainIdsStr = undeployedChainIds.join(', ');
+
+      if (undeployedChainIds.length > 0) {
+        log.warning(
+          `Faucet module unavailable on chain "${undeployedChainIdsStr}" in the "${networkConfig.network}" network.`,
+        );
+
+        const { deployFaucet } = await option.deployFaucet();
+
+        if (!deployFaucet) {
+          return;
+        }
+        const loader = ora(
+          `Deploying faucet on chain Id(s): "${undeployedChainIdsStr}" in "${network}" network...\n`,
+        ).start();
+
+        await deployDevNetFaucet(undeployedChainIds).catch((e) => {
```

Comment:
I would map over `undeployedChainIds` in here so you can handle success/error individually per chain, and if 1 deployment fails but others succeed you can give the correct output information.